### PR TITLE
pkp/pkp-lib#9857 FieldSlider

### PR DESCRIPTION
**Pull request by jardakotesovec**

### CODE REVIEW

**Review by touhidurabir — CHANGES_REQUESTED** [review]
Along with all these control option , I think we should also add the ability to define what is the range steps would be . It should not be a nullable and allow both int and float value .

**Review by touhidurabir — CHANGES_REQUESTED** [review]
I think we should allow both int and float value here and also below for max value . There is no saying that it will always be integer so better to have that flexibility .

File: classes/components/forms/FieldSlider.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    public function getConfig()
+    {
+        $config = parent::getConfig();
+        $config['min'] = $this->min;
+        $config['max'] = $this->max;
+        $config['minLabel'] = $this->minLabel;
+        $config['maxLabel'] = $this->maxLabel;
+        $config['valueLabel'] = $this->valueLabel;
+        $config['valueLabelMin'] = $this->valueLabelMin;
+        $config['valueLabelMax'] = $this->valueLabelMax;
```
Comment (touhidurabir) [inline classes/components/forms/FieldSlider.php:73]: Along with all these control option , I think we should also add the ability to define what is the range steps would be . It should not be a nullable and allow both int and float value .

File: classes/components/forms/FieldSlider.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+class FieldSlider extends Field
+{
+    /** @copydoc Field::$component */
+    public $component = 'field-slider';
+
+
+    /**
+     * Range min value
+     */
+    public int $min;
```
Comment (touhidurabir) [inline classes/components/forms/FieldSlider.php:27]: Along with all these control option , I think we should also add the ability to define what is the range steps would be . It should not be a nullable and allow both int and float value .

**Comment by touhidurabir** [comment]
@jardakotesovec I think all ok for now as adding too much option right away make no sense unless we really need that . Also as this going into the main only, we have enough time to update/modify if necessary . Just few minor enhancements to address .

File: classes/components/forms/FieldSlider.php (hunk 1 above)
Comment (jardakotesovec) [reply classes/components/forms/FieldSlider.php:73]: @touhidurabir On the other hand - the improvements you are suggesting are small effort..  so not really problem include them. Will do the int/float & step property.

Comment (touhidurabir) [reply classes/components/forms/FieldSlider.php:73]: As the value can be float and in such case steps can be float also . Right now we do no have a use case but it's not unnatural to have the slider range increase like [0.5, 1.0, 1.5, 2.0, 2.5, ...] .

**Comment by jardakotesovec** [comment]
@touhidurabir Updated - can you double check it looks ok? Thanks!

**Comment by touhidurabir** [comment]
@jardakotesovec all seems ok , good to merge as I see it .

**Pull request merged**